docs/source/runtime_troubles.rst

```diff
@@ -0,0 +1,311 @@
+.. _runtime_troubles:
+
+Annotation issues at runtime
+============================
+
+Idiomatic use of type annotations can sometimes run up against what a given
+version of Python considers legal code. This section describes these scenarios
+and explains how to get your code running again. Generally speaking, we have
+three tools at our disposal:
+
+* For Python 3.7 through 3.9, use of ``from __future__ import annotations``
+  (:pep:`563`), made the default in Python 3.10 and later
```

Comment:
Nit: Since 3.10 is not out yet, this would be better written as "... will be the default in Python 3.10 and later".